### PR TITLE
Make it a real object

### PR DESCRIPTION
**Pull request by roebuk**

### CODE REVIEW

**Review by Briggy2501 — APPROVED** [review]
Recreated this change locally to test recaptcha, worked fine for me. LGTM

File: snowpenguin/django/recaptcha2/templates/snowpenguin/recaptcha/recaptcha_init.html [hunk 1]
```diff
@@ -1,12 +1,13 @@
 <script src="https://www.google.com/recaptcha/api.js{% if explicit %}?render=explicit{% if language %}&hl={{ language }}{% endif %}{% elif language %}?hl={{ language }}{% endif %}" async defer></script>
 {% if explicit %}
 <script>
-    window.addEventListener('load', function() {
```
Comment (SammyM) [inline snowpenguin/django/recaptcha2/templates/snowpenguin/recaptcha/recaptcha_init.html]: Put this indentation back please.

File: snowpenguin/django/recaptcha2/templates/snowpenguin/recaptcha/recaptcha_init.html [hunk 2]
```diff
@@ -1,12 +1,13 @@
 <script src="https://www.google.com/recaptcha/api.js{% if explicit %}?render=explicit{% if language %}&hl={{ language }}{% endif %}{% elif language %}?hl={{ language }}{% endif %}" async defer></script>
 {% if explicit %}
 <script>
```
Comment (SammyM) [inline snowpenguin/django/recaptcha2/templates/snowpenguin/recaptcha/recaptcha_init.html]: Indent this within the Django `if` please.

File: snowpenguin/django/recaptcha2/templates/snowpenguin/recaptcha/recaptcha_init.html [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
-      var captchas = [].slice.call(document.querySelectorAll('.g-recaptcha'));
+window.addEventListener('load', function() {
+  var captchas = [].slice.call(document.querySelectorAll('.g-recaptcha'));
 
-      captchas.map(function(el) {
-        grecaptcha.render(el.id, el.dataset);
-      });
-    });
+  captchas.forEach(function(el) {
+    var id = grecaptcha.render(el.id, Object.assign({}, el.dataset));
```
Comment (SammyM) [inline snowpenguin/django/recaptcha2/templates/snowpenguin/recaptcha/recaptcha_init.html]: `Id` isn't a very descriptive variable name.

File: snowpenguin/django/recaptcha2/templates/snowpenguin/recaptcha/recaptcha_init.html [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
+    el.setAttribute('data-captchaId', id);
```
Comment (SammyM) [inline snowpenguin/django/recaptcha2/templates/snowpenguin/recaptcha/recaptcha_init.html]: I'm pretty sure this can just be `el.dataset.captchaId  = id;`

**Review by SammyM — CHANGES_REQUESTED** [review]

File: snowpenguin/django/recaptcha2/templates/snowpenguin/recaptcha/recaptcha_init.html (hunk 3 above)
Comment (SammyM) [inline snowpenguin/django/recaptcha2/templates/snowpenguin/recaptcha/recaptcha_init.html]: `Object.assign` is not supported in IE11.

**Review by SammyM — APPROVED** [review]

**Pull request merged**